examples/Bunch/include/simulation_defines/param/fieldBackground.param

```diff
@@ -0,0 +1,222 @@
+         *
+         * \param cellIdx The total cell id counted from the start at t=0
+         * \param currentStep The current time step */
+        HDINLINE float3_X
+        operator()( const DataSpace<simDim>& cellIdx,
+                    const uint32_t currentStep ) const
+        {
+            /* example 1: TWTS background pulse */
+            /** unit: meter */
+            const double WAVE_LENGTH_SI = 0.8e-6;
+            
+            /** UNITCONV */
+            /* const double UNITCONV_Intens_to_A0 = ::picongpu::SI::ELECTRON_CHARGE_SI
+                * ::picongpu::SI::ELECTRON_CHARGE_SI * 2.0 * WAVE_LENGTH_SI * WAVE_LENGTH_SI
+                / (4.0 * PI * PI * ::picongpu::SI::ELECTRON_MASS_SI
+                * ::picongpu::SI::ELECTRON_MASS_SI * ::picongpu::SI::SPEED_OF_LIGHT_SI
+                * ::picongpu::SI::SPEED_OF_LIGHT_SI * ::picongpu::SI::SPEED_OF_LIGHT_SI
+                * ::picongpu::SI::SPEED_OF_LIGHT_SI * ::picongpu::SI::SPEED_OF_LIGHT_SI
+                * ::picongpu::SI::EPS0_SI); */
+            const double UNITCONV_A0_to_Amplitude_SI = -2.0 * PI / WAVE_LENGTH_SI
+                * ::picongpu::SI::ELECTRON_MASS_SI * ::picongpu::SI::SPEED_OF_LIGHT_SI
+                * ::picongpu::SI::SPEED_OF_LIGHT_SI / ::picongpu::SI::ELECTRON_CHARGE_SI;
+
+            /** unit: W / m^2 */
+            /* const double _PEAK_INTENSITY_SI = 3.4e19 * 1.0e4; */
+            /** unit: none */
+            /* const double _A0  = _PEAK_INTENSITY_SI * UNITCONV_Intens_to_A0; */
+
+            /** unit: none */
+            const double _A0  = 1.0;
+
+            /** unit: Volt /meter */
+            const float3_64 invUnitField = float3_64(1.0/unitField[0],
+                             1.0/unitField[1],
+                             1.0/unitField[2]);
+            
+            const float3_X AMPLITUDE_SI = precisionCast<float_X>(
+                                float_64(_A0 * UNITCONV_A0_to_Amplitude_SI) * invUnitField );
+
+            /** unit: Volt /meter */
+            /* const double AMPLITUDE_SI = 1.738e13; */
+            
+            /* Note: twtsFieldE(...) is normalized, such that peak amplitude equals unity. */
+            return AMPLITUDE_SI * twtsFieldE( cellIdx, currentStep );
+        }
+    };
+
+    class FieldBackgroundB
+    {
+    public:
+        /* Add this additional field for pushing particles */
+        static const bool InfluenceParticlePusher = PARAM_INCLUDE_FIELDBACKGROUND;
+
+        /* TWTS B-fields need to be initialized on host,
+         * so they can look up global grid dimensions. */
+        templates::TWTSFieldB twtsFieldB;
+        
+        /* We use this to calculate your SI input back to our unit system */
+        const float3_64 unitField;
+        HINLINE FieldBackgroundB( const float3_64 unitField ) : unitField(unitField),
+        
+            twtsFieldB( 30.0e-6,  /* focus_y [m], the distance to the laser focus in y-direction */
```

Comment:
please move your comment above the number to be consistent with the following lines.